src/operators/Remap/avtRemapFilter.C

```diff
@@ -131,447 +138,533 @@ avtRemapFilter::SetAtts(const AttributeGroup *a)
-        // over the subcells in that grid, calculate the volumes, and generate
-        // A new volume for the rectilinear grid. We also need the ratio
-        // between old and new volumes to weigh the variables (like mass, density,
-        // pressure, etc.).
-        
-        vtkDoubleArray* subCellVolumes = CalculateCellVolumes(ug, "subCellVolume");
-        
-// --- DEBUG --- DEBUG --- DEBUG --- DEBUG --- DEBUG --- DEBUG --- DEBUG ---- //
-        double DEBUG_rCellVolumeTEST = 0.0;
-        for (vtkIdType j = 0; j < subCellVolumes->GetNumberOfTuples(); j++) {
-            DEBUG_rCellVolumeTEST += subCellVolumes->GetComponent(j,0);
-        }
-        if (DEBUG_rCellVolumeTEST != rCellVolume)
-        {
-            double diff = std::abs(DEBUG_rCellVolumeTEST - rCellVolume);
-            DEBUG_maxDiff = diff > DEBUG_maxDiff ? diff : DEBUG_maxDiff;
-            std::cout << "Volume from subcells: " << DEBUG_rCellVolumeTEST << std::endl;
-        }
-// --- DEBUG --- DEBUG --- DEBUG --- DEBUG --- DEBUG --- DEBUG --- DEBUG ---- //
-        
-        // Get the volume of the original cells for this ugrid
-        vtkDoubleArray* originalCellVolumes = vtkDoubleArray::SafeDownCast(
-            ug->GetCellData()->GetArray("avtRemapOriginalVolume"));
-        
-        
-        
-        // --- Calculate variable updates --- //
-        // Two types of variables: intrinsic (like density) and extrinsic (like
-        // mass). To update the variables:
-        //        (1) Intrinsic values must be made extrinsic within the volume of
-        //            the sub cell, then totaled among the set of sub cells, then
-        //            made intrinsic again within the volume of the new cell.
-        //        (2) Extrinsic values must be made intrisic within the volume of
-        //            original cell, then made extrinsic within the volume of the
-        //            sub cell, then totaled among the set of sub cells.
-        
-        for (int vdx = 0; vdx < nVariables; vdx++)
+    // --- Loop over each cell --- //
+    debug5 << "Beginning the clipping loop" << std::endl;
+    for (int k = 0; k < (is3D ? atts.GetCellsZ() : 1); ++k) // Loop over z-dim
+    {
+        for (int j = 0; j < atts.GetCellsY(); ++j) // Loop over y-dim
         {
-            double value = 0.0;
-            vtkDataArray* myVariable = ug->GetCellData()->GetArray(vdx);
-                // vtkDoubleArray::SafeDownCast(ug->GetCellData()->GetArray(vdx));
-            if (varType == RemapAttributes::intrinsic) // like density
+            for (int i = 0; i < atts.GetCellsX(); ++i) // Loop over i-dim
             {
-                for (vtkIdType tuple = 0;
-                     tuple < myVariable->GetNumberOfTuples(); tuple++)
+                // Get the index for rg
+                int rCell = i +
+                        atts.GetCellsX() * j +
+                        atts.GetCellsX() * atts.GetCellsY() * k;
+
+                // --- Clip the cell --- //
```

Comment:
If you instantiate (and delete) these clippers outside of the for loops, it should speed things up a bit. Using RemoveAllInputs/RemoveAllInputConnections will usually let you "reset" vtk objects like this so that they can be re-used with new data and new settings without having to re-create them.

Reply:
Thanks for the tip. I knew I was taking a performance hit by doing that, but I was unaware of the ability to reset the clippers at each loop. Fixed.